File: packages/android/app/src/main/java/com/bookpost/domain/model/ReadingHistory.kt
package com.bookpost.domain.model

data class ReadingHistoryEntry(
    val id: Int,
    val userId: Int? = null,
    val itemType: ItemType,
    val itemId: Int,
    val title: String? = null,
    val coverUrl: String? = null,
    val lastPage: Int? = null,
    val lastReadAt: String? = null,
    val createdAt: String? = null
)

enum class ItemType(val value: String) {
    EBOOK("ebook"),
    MAGAZINE("magazine"),
    BOOK("book");

    companion object {
        fun fromString(value: String): ItemType {
            return entries.find { it.value == value } ?: EBOOK
        }
    }
}

File: packages/ios/BookPost/Models/ReadingHistory.swift
import Foundation

enum ItemType: String, Codable {
    case ebook
    case magazine
    case book
}

struct ReadingHistoryEntry: Codable, Identifiable {
    let id: Int
    let userId: Int?
    let itemType: String
    let itemId: Int
    let title: String?
    let coverUrl: String?
    let lastPage: Int?
    let lastReadAt: String?
    let createdAt: String?

    var type: ItemType {
        ItemType(rawValue: itemType) ?? .ebook
    }
}

struct ReadingHistoryResponse: Codable {
    let data: [ReadingHistoryEntry]
}

struct UpdateReadingHistoryRequest: Codable {
    let itemType: String
    let itemId: Int
    let title: String?
    let coverUrl: String?
    let lastPage: Int?
}
